pyface/undo/tests/test_command_stack.py

```diff
@@ -31,21 +34,92 @@ def test_empty_command_stack(self):
 
     def test_1_command_pushed(self):
         with self.assert_n_commands_pushed(self.stack, 1):
-            self.stack.push(self.command)
+            with self.assertTraitChanges(self.stack.undo_manager,
+                                         'stack_updated', count=1):
+                self.stack.push(self.command)
 
     def test_n_command_pushed(self):
         n = 4
         with self.assert_n_commands_pushed(self.stack, n):
             for i in range(n):
+                with self.assertTraitChanges(self.stack.undo_manager,
+                                            'stack_updated', count=1):
+                    self.stack.push(self.command)
+
+    def test_push_after_undo(self):
+        with self.assert_n_commands_pushed(self.stack, 1):
+            self.stack.push(self.command)
+            self.stack.undo()
+            with self.assertTraitChanges(self.stack.undo_manager,
+                                         'stack_updated', count=1):
+                self.stack.push(self.command)
+
+    def test_push_after_n_undo(self):
+        with self.assert_n_commands_pushed(self.stack, 1):
+            n = 4
+            for i in range(n):
+                self.stack.push(self.command)
+            n = 4
+            for i in range(n):
+                self.stack.undo()
+    
+            with self.assertTraitChanges(self.stack.undo_manager,
+                                         'stack_updated', count=1):
                 self.stack.push(self.command)
 
+    # Command merging tests ---------------------------------------------------
+
+    def test_1_merge_command_pushed(self):
+        self.command = MergeableCommand()
+        with self.assert_n_commands_pushed(self.stack, 1):
+            with self.assertTraitChanges(self.stack.undo_manager,
+                                         'stack_updated', count=1):
+                self.stack.push(self.command)
+
+    def test_n_merge_command_pushed(self):
+        n = 4
+        with self.assert_n_commands_pushed(self.stack, 1):
+            self.command = MergeableCommand()
+            self.stack.push(self.command)
+            for i in range(n):
+                command = MergeableCommand()
+                with self.assertTraitChanges(self.stack.undo_manager,
+                                         'stack_updated', count=1):
```

Comment:
nit-picky suggestion

```suggestion
                with self.assertTraitChanges(self.stack.undo_manager,
                                             'stack_updated', count=1):
```